src/dcos_e2e/cluster.py

```diff
@@ -116,9 +117,14 @@ def wait_for_dcos(self) -> None:
             '--diag',
         ]
 
+        # Must be run privileged
+        if not self.default_ssh_user == 'root':
+            diagnostics_args = ['sudo'] + diagnostics_args
```

Comment:
At least for now this code isn't covered by tests, so IMO remove it and we can discuss it for the follow-up.